src/main/java/org/humancellatlas/ingest/project/Project.java

```diff
@@ -67,4 +64,18 @@ public SubmissionEnvelope getOpenSubmissionEnvelope(){
         }
         return null;
     }
-}
\ No newline at end of file
+
+    @JsonIgnore
+    public boolean removeSubmissionEnvelope(SubmissionEnvelope submissionEnvelope, boolean forceRemoval) {
+        if(!(submissionEnvelope.isOpen() || forceRemoval))
+            throw new UnsupportedOperationException("Cannot remove submission from Project if it is already submitted!");
+        boolean success = true;
```

Comment:
Can do `boolean success = this.supplementaryFiles.removeIf(file -> file.getSubmissionEnvelope().equals(submissionEnvelope));
` here and remove the for loop here. Also I'm not 100% but the loop below might throw a ConcurrentModificationException